server/src/internalClusterTest/java/org/elasticsearch/cluster/ClusterHealthIT.java

```diff
@@ -309,14 +310,26 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
 
     public void testHealthOnMasterFailover() throws Exception {
         final String node = internalCluster().startDataOnlyNode();
+        boolean withIndex = randomBoolean();
+        if (withIndex) {
+            // create index with many shards to provoke the health request to wait (for green) while master is being shut down.
+            createIndex("test", Settings.builder().put(IndexMetadata.SETTING_NUMBER_OF_REPLICAS, randomIntBetween(0, 10)).build());
```

Comment:
I'm confused. How can the cluster ever be green with that many replicas? Should this be `number_of_shards`?

Reply:
It ensures that the cluster is yellow when the health request is made, making the health request wait on the observer, triggering the call to `onClusterServiceClose` when master is shutdown.

The number of replicas is [cleared to 0](https://github.com/elastic/elasticsearch/pull/61709/files#diff-fb73466f78397039bcf00e8037472624R330) after having fired all the async restarts and done the master restarts. That ensures that all the requests responds with green status.

Reply:
ah ok, can you add a comment to that effect